### ISSUE TITLE
false positive for valid pattern matching in Lint/Syntax

### ISSUE THREAD

**Issue by timoschilling** [rubocop-hq/rubocop#8165]
## Expected behavior

No offence is detected

## Actual behavior

Unexpected `Lint/Syntax` detections

## Steps to reproduce the problem

Run rubocop on a file with this content:

```ruby
{ Foo: 1 } in { Foo: foo }
{ Foo: { bar: 1 }} in { Foo: { bar: } }
```

## Result

```
Offenses:

test.rb:1:17: E: Lint/Syntax: formal argument cannot be a constant
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
{ Foo: 1 } in { Foo: foo }
                ^^^^
test.rb:2:24: E: Lint/Syntax: formal argument cannot be a constant
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
{ Foo: {bar: 1 }} in { Foo: { bar: } }
                       ^^^^

12 files inspected, 2 offenses detected
```

## Annotation

This occurrence should only be shown for

```ruby
{ Foo: 1 } in { Foo: }
```

## RuboCop version

```
0.85.1 (using Parser 2.7.1.2, rubocop-ast 0.0.3, running on ruby 2.7.0 x86_64-darwin19)
```

**Comment by koic** [comment]
I have opened https://github.com/whitequark/parser/pull/717 and merged. It will be resolved in the next release of Parser.
